networkx/algorithms/centrality/group.py

```diff
@@ -93,40 +100,134 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None):
        Group Centrality Maximization via Network Design.
        SIAM International Conference on Data Mining, SDM 2018, 126–134.
        https://sites.cs.ucsb.edu/~arlei/pubs/sdm18.pdf
+    .. [4] Rami Puzis, Yuval Elovici, and Shlomi Dolev.
+       "Fast algorithm for successive computation of group betweenness centrality."
+        https://journals.aps.org/pre/pdf/10.1103/PhysRevE.76.056709?casa_token=YvH6xTIc6AUAAAAA%3A90BkxMRiQyOg9y7PEU_lrcf1FXPF_sqTCI_Y_2lD0QodSeyXAQtz-s0_e1Z0CtkZ1bWdVCkK9S5fXw
+
     """
-    betweenness = 0  # initialize betweenness to 0
-    V = set(G)  # set of nodes in G
-    C = set(C)  # set of nodes in C (group)
-    if len(C - V) != 0:  # element(s) of C not in V
-        raise nx.NodeNotFound(
-            "The node(s) " + str(list(C - V)) + " are not " "in the graph."
-        )
-    V_C = V - C  # set of nodes in V but not in C
-    # accumulation
-    for pair in combinations(V_C, 2):  # (s, t) pairs of V_C
-        try:
-            paths = 0
-            paths_through_C = 0
-            for path in nx.all_shortest_paths(
-                G, source=pair[0], target=pair[1], weight=weight
-            ):
-                if set(path) & C:
-                    paths_through_C += 1
-                paths += 1
-            betweenness += paths_through_C / paths
-        except nx.exception.NetworkXNoPath:
-            betweenness += 0
-    # rescaling
-    v, c = len(G), len(C)
-    if normalized:
-        scale = 1 / ((v - c) * (v - c - 1))
-        if not G.is_directed():
-            scale *= 2
+    GBC = []  # initialize betweenness
+    list_of_groups = True
+    #  check weather C contains one or many groups
+    if any(el in G for el in C):
+        C = [C]
+        list_of_groups = False
+    set_v = {node for group in C for node in group}
+    if set_v - G.nodes:  # element(s) of C not in G
+        raise nx.NodeNotFound(f"The node(s) {set_v - G.nodes} are in C but not in G.")
+
+    # pre-processing
+    PB, sigma, D = _group_preprocessing(G, set_v, weight)
+
+    # the algorithm for each group
+    for group in C:
+        group = set(group)  # set of nodes in group
+        # initialize the matrices of the sigma and the PB
+        GBC_group = 0
+        sigma_m = deepcopy(sigma)
+        PB_m = deepcopy(PB)
+        sigma_m_v = deepcopy(sigma_m)
+        PB_m_v = deepcopy(PB_m)
+        for v in group:
+            GBC_group += PB_m[v][v]
+            for x in group:
+                for y in group:
+                    dxvy = 0
+                    dxyv = 0
+                    dvxy = 0
+                    if not (
+                        sigma_m[x][y] == 0 or sigma_m[x][v] == 0 or sigma_m[v][y] == 0
+                    ):
+                        if D[x][v] == D[x][y] + D[y][v]:
+                            dxyv = 1.0 * sigma_m[x][y] * sigma_m[y][v] / sigma_m[x][v]
```

Comment:
Why do we have the `1.0 * ` in these expressions?

Reply:
I don't need that. Deleted